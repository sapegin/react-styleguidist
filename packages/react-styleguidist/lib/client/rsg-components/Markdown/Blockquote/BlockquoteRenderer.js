import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import Styled from 'rsg-components/Styled';

const styles = function styles(_ref) {
  const space = _ref.space;

      
const color = _ref.color;

      
const fontSize = _ref.fontSize;

      
const fontFamily = _ref.fontFamily;
  return {
    blockquote: {
      margin: [[space[2], space[4]]],
      padding: 0,
      color: color.base,
      fontFamily: fontFamily.base,
      fontSize: fontSize.base,
      lineHeight: 1.5
    }
  };
};

export function BlockquoteRenderer(_ref2) {
  const classes = _ref2.classes;

      
const className = _ref2.className;

      
const children = _ref2.children;
  const blockquoteClasses = cx(classes.blockquote, className);
  return React.createElement("blockquote", {
    className: blockquoteClasses
  }, children);
}
BlockquoteRenderer.propTypes = {
  classes: PropTypes.object.isRequired,
  className: PropTypes.string,
  children: PropTypes.node.isRequired
};
export default Styled(styles)(BlockquoteRenderer);