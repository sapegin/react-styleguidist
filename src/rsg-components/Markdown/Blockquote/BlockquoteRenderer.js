import React from 'react';
import PropTypes from 'prop-types';

import Styled from 'rsg-components/Styled';

const styles = ({ space, color, fontSize, fontFamily }) => ({
	blockquote: {
		margin: [[space[2], space[4]]],
		padding: 0,
		color: color.base,
		fontFamily: fontFamily.base,
		fontSize: fontSize.base,
		lineHeight: 1.5,
	},
});

export function BlockquoteRenderer({ classes, children }) {
	return <blockquote className={classes.blockquote}>{children}</blockquote>;
}
BlockquoteRenderer.propTypes = {
	classes: PropTypes.object.isRequired,
	children: PropTypes.node.isRequired,
};

export default Styled(styles)(BlockquoteRenderer);
