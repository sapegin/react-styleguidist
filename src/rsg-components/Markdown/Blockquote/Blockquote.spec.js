import React from 'react';

import Blockquote from './index';

describe('Markdown Blockquote', () => {
	it('should render a blockquote', () => {
		const actual = render(<Blockquote>To be, or not to be: that is the question</Blockquote>);

		expect(actual).toMatchSnapshot();
	});
});
