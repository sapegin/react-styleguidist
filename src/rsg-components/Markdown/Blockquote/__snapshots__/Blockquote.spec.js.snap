// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Markdown Blockquote should render a blockquote 1`] = `
<blockquote
  class="rsg--blockquote-0"
>
  To be, or not to be: that is the question
</blockquote>
`;
